ipykernel/kernelspec.py

```diff
@@ -118,9 +122,13 @@ def install(kernel_spec_manager=None, user=False, kernel_name=KERNEL_NAME, displ
         overrides = dict(display_name=display_name)
     else:
         overrides = None
-    path = write_kernel_spec(overrides=overrides)
-    dest = kernel_spec_manager.install_kernel_spec(path,
-        kernel_name=kernel_name, user=user, prefix=prefix)
+    if profile:
+        extra_arguments = ["--profile", profile]
```

Comment:
I don't think the display_name horizontal space is so precious that `profile=` should be omitted, but feel free to try both and see how it feels. I don't think [name] is sufficiently clear, since it could be indicating other things, such as conda/virtual env, etc.